pyrate/core/phase_closure/mst_closure.py

```diff
@@ -82,29 +84,40 @@ def edges(self):
         return [Edge(swe.first, swe.edge.second) for swe in self.loop]
 
 
-def __discard_cycles_with_same_members(simple_cycles: List[List[date]]) -> List[List[date]]:
-    seen_sc_sets = set()
-    filtered_sc = []
-    for sc in simple_cycles:
-        loop = sc[:]
-        sc.sort()
-        sc = tuple(sc)
-        if sc not in seen_sc_sets:
-            seen_sc_sets.add(sc)
-            filtered_sc.append(loop)
-    return filtered_sc
+def __discard_cycles_with_same_members(cycles: List[List[date]]) -> List[List[date]]:
+    log.debug(f"Discarding loops that contain the same set of edges as another loop")
+    seen_cycle_sets = set()
+    filtered_cycles = []
+    for c in cycles:
+        loop = c[:]
+        c.sort()
+        c = tuple(c)
+        if c not in seen_cycle_sets:
+            seen_cycle_sets.add(c)
+            filtered_cycles.append(loop)
+    log.debug(f"Number of remaining loops is {len(filtered_cycles)}")
+    return filtered_cycles
```

Comment:
I am not convinced we need a separate function here. It is not being called in any tests.

I suggest merge this code in to the `__find_closed_loops` function, to avoid having too many superfluous functions